gui/quantum.lua

```diff
@@ -11,262 +11,212 @@ local quickfort = reqscript('quickfort')
-                return
-            end
-
-            self:dismiss()
-            self:commit(pos, qsp_pos)
-        end
-    end
-end
+--UI Data Tables
+qspOutputHor = {
+	["North"]=0,
+	["East"]=1,
+	["South"]=0,
+	["West"]=-1
+}
 
-local function get_feeder_pos(feeder_tiles)
-    for z,rows in pairs(feeder_tiles) do
-        for y,row in pairs(rows) do
-            for x in pairs(row) do
-                return xyz2pos(x, y, z)
-            end
-        end
-    end
-end
+qspOutputVert = {
+	["North"]=-1,
+	["East"]=0,
+	["South"]=1,
+	["West"]=0
+}
 
-local function get_moves(move, move_back, start_pos, end_pos,
-                         move_to_greater_token, move_to_less_token)
-    if start_pos == end_pos then
-        return move, move_back
-    end
-    local diff = math.abs(start_pos - end_pos)
-    local move_to_greater_pattern = ('{%s %%d}'):format(move_to_greater_token)
-    local move_to_greater = move_to_greater_pattern:format(diff)
-    local move_to_less_pattern = ('{%s %%d}'):format(move_to_less_token)
-    local move_to_less = move_to_less_pattern:format(diff)
-    if start_pos < end_pos then
-        return move..move_to_greater, move_back..move_to_less
-    end
-    return move..move_to_less, move_back..move_to_greater
-end
+qspTrackstopDir = {
+	["North"]='trackstopN',
+	["East"]='trackstopE',
+	["South"]='trackstopS',
+	["West"]='trackstopW'
+}
 
-local function get_quantumstop_data(dump_pos, feeder_pos, name)
-    local move, move_back = get_moves('', '', dump_pos.z, feeder_pos.z, '<','>')
-    move, move_back = get_moves(move, move_back, dump_pos.y, feeder_pos.y,
-                                'Down', 'Up')
-    move, move_back = get_moves(move, move_back, dump_pos.x, feeder_pos.x,
-                                'Right', 'Left')
+qspOutputType = {
+	[true]='afunswebhlzSgpdyr',
+	[false]='afunswebhlzSgpd'
+}
 
-    local quantumstop_name_part, quantum_name_part = '', ''
-    if name ~= '' then
-        quantumstop_name_part = (' name="%s quantum"'):format(name)
-        quantum_name_part = ('{givename name="%s dumper"}'):format(name)
-    end
 
-    return ('{quantumstop%s move="%s" move_back="%s"}%s')
-           :format(quantumstop_name_part, move, move_back, quantum_name_part)
-end
 
-local function get_quantum_data(name)
-    local name_part = ''
-    if name ~= '' then
-        name_part = (' name="%s"'):format(name)
-    end
-    return ('{quantum%s}'):format(name_part)
+--Actually making the QSP
+function QuantumUI:commit()
+	name = self.subviews.name.text
+    dump_dir = self.subviews.dir:getOptionLabel()
+    allow_refuse = self.subviews.refuse:getOptionValue()
+	input_id = dfhack.gui.getSelectedStockpile().id
+	output_hor = qspOutputHor[dump_dir]
+	output_vert = qspOutputVert[dump_dir]
+	output_type = qspOutputType[allow_refuse]
+	trackstop_dir = qspTrackstopDir[dump_dir]
+	if
+		name == ''
+	then
+		name = 'Unnamed QSP'
+	end
+	placed_output = output_type..'{quantum=true}'
```

Comment:
should be of the form:
`ry{name="$name quantum" quantum=true}:+all`
or
`c{name="$name quantum" quantum=true}:+all`